wallet/config.go

```diff
@@ -0,0 +1,38 @@
+/*
+ * Copyright (c) 2013, 2014 Conformal Systems LLC <info@conformal.com>
+ *
+ * Permission to use, copy, modify, and distribute this software for any
+ * purpose with or without fee is hereby granted, provided that the above
+ * copyright notice and this permission notice appear in all copies.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+ * WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+ * MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+ * ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+ * WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+ * ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+ * OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+ */
+
+package wallet
+
+import (
+	"github.com/btcsuite/btcd/chaincfg"
+)
+
+const (
+	walletDbWatchingOnlyName = "wowallet.db"
+)
+
+type Config struct {
+	DataDir          string
+	WalletPass       string
+	DisallowFree     bool
+	UseAddrIndex     bool
+	ChainParams		 *chaincfg.Params
+	WalletDbName     string
+}
+
+func (w *Wallet) GetConfig() (*Config) {
```

Comment:
I would prefer if the config was just an exported member of the wallet, so it could be accessed directly without a getter.